chain/client/src/stateless_validation/partial_witness/partial_witness_actor.rs

```diff
@@ -215,7 +215,7 @@ impl PartialWitnessActor {
         let encode_timer = metrics::PARTIAL_WITNESS_ENCODE_TIME
             .with_label_values(&[shard_id_label.as_str()])
             .start_timer();
-        let validator_witness_tuple = self.generate_state_witness_parts(
+        let mut validator_witness_tuple = self.generate_state_witness_parts(
```

Comment:
Passing the chunk_validators vec to this function is starting to look dangerous to me now. We really really care about the order of the chunk validators as part owners are determined that way, and over time we are seeing ordering change.

Would it be possible to call our beloved `.ordered_chunk_validators()` function within `generate_state_witness_parts`?

Reply:
For example, it would be a disaster is some future code were to say remove self from `chunk_validators` before this function call.

Reply:
I did some refactorings to address your comment:

1. Moved the computation of chunk_validators from `handle_distribute_state_witness_request` to `generate_state_witness_parts`.
2. Moved the recording of the witness sent event to `send_state_witness_parts` (since it needs the number of validators witness is sent). It is better now because it actually captures the right number of validators now (`chunk_validators` except for self).
3. Computing `chunk_validators` in `forward_state_witness_part` separately. This is safer instead of relying the caller and simplifies the other caller of the same function.

@shreyan-gupta please take another look.